elip-ct-descriptors.mediawiki

```diff
@@ -0,0 +1,167 @@
+
+===Motivation===
+
+Confidential Transactions involve the use of uniformly random "blinding factors" associated to every confidential output.
+These random values are secret, but must be known by the sender (in order to construct a transaction) as well as recipient (in order to recognize the received assets and amounts, and to spend the outputs).
+This means they must be chosen by the sender of a transaction and somehow securely communicated to the recipient.
+
+The way this is done is that confidential addresses contain the public key of an extra "blinding key" pair created by the recipient.
+When blinding, the sender chooses a fresh EC Diffie-Hellman (ECDH) private key, encodes the corresponding public key in the "nonce" field of the transaction, derives an ECDH secret between the fresh private key and the public blinding key, uses the ECDH secret to encrypt the blinding factors for the output, and encodes the encrypted result in the rangeproof.
+The recipient, when recognizing a scriptPubKey corresponding to the ordinary part of her confidential address, uses the private blinding key for that address in conjunction with the nonce field of the txout to re-derive the ECDH secret and decrypt the blinding factors.
+
+This document defines a standard way for the recipient's wallet to compute blinding key pairs. There are a number of requirements:
+
+* Confidential addresses, including their attached blinding key, should be derivable from a single descriptor without extra data
+* Wallets should be able to choose the granularity of their blinding keys, so that the revelation of private blinding keys may unblind one, a subset, or all, of its blinded outputs
+* In multiparty settings, each wallet should be able to restrict this granularity
+* It should be possible somehow to do public derivation of CT addresses, given only a descriptor containing (extended) public keys
+
+The current most popular scheme, [https://github.com/satoshilabs/slips/blob/master/slip-0077.md SLIP-77], does not satisfy any of these criteria, which limits its usefulness as we move toward a descriptor-centric setting in which multiparty addresses are common.
+
+==Design==
+
+===Overview===
+We propose a new <code>ct</code> descriptor which wraps any other descriptor type in the form <code>ct(<BLINDING_KEY>, <DESCRIPTOR>)</code>.
+Here <code>DESCRIPTOR</code> refers to any existing descriptor, e.g. <code>elwsh(...)</code> or <code>eltr(...)</code> and <code>BLINDING_KEY</code> is a new expression which must be one of
+* <code>slip77(<64-character hex>)</code> which indicates that blinding keys for these addresses are derived via SLIP77; or
+* <code><KEY></code> which is an ordinary (extended) public key; the blinding key is derived by a Taproot-style tweak of the key with the data of <code><DESCRIPTOR></code>.
+
+In the <code><KEY></code> expression, all key(s) must be compressed; x-only and uncompressed keys are invalid.
```

Comment:
nit: It would be great if this specifically mentions `KEY` expression in descriptor spec. Reading the words compressed ordinary key confused me a bit.
 
another nit: This also mentions multi-path KEY expressions. I don't think we want to support that.

Reply:
Where does it mention multi-path KEY expressions? I did not even know those existed when I was writing this.

Reply:
Perhaps the "all key(s)" is confusing. There's just a single key.

Reply:
It might be a `musig` of multiple keys.

Reply:
in the `<KEY>` expression?

Reply:
Key expressions are defined as follows. `https://github.com/bitcoin/bips/blob/master/bip-0380.mediawiki#key-expressions`. I think what we mean to say that all `KEY` expressions that are not uncompressed keys are allowed.

Reply:
I'm not sure. We say `<code><KEY></code> which is an ordinary (extended) public key`. We don't state anywhere that this is a "key expression" as in BIP380. 

If it's supposed to be a key expression as in BIP380, we should say this. If not, we should consider renaming `KEY`.

Reply:
Ok, I will reword this. Yes, it is supposed to be a key expression.

Reply:
Though it would be nice if we had our own ELIP for key expressions so that we can extend it to support `musig()` eventually. Though I have zero interest in doing that. This ELIP has been more than enough standards work for one lifetime.